pkg/apiserver/apiserver.go

```diff
@@ -55,13 +55,12 @@ func init() {
 	metrics.Register()
 }
 
-// monitorFilter creates a filter that reports the metrics for a given resource and action.
-func monitorFilter(action, resource string) restful.FilterFunction {
-	return func(req *restful.Request, res *restful.Response, chain *restful.FilterChain) {
-		reqStart := time.Now()
-		chain.ProcessFilter(req, res)
-		httpCode := res.StatusCode()
-		metrics.Monitor(&action, &resource, util.GetClient(req.Request), &httpCode, reqStart)
+type instrumenterFunc func(routeFunc restful.RouteFunction) restful.RouteFunction
```

Comment:
Reason was to create the instrumenter func only once, but again only negligible benefits at API installer time rather than normal runtime. Changing.